### ISSUE TITLE
Cant save txt when stream is set to True

### ISSUE THREAD

**Issue by mohamad-tohidi** [ultralytics/ultralytics#10079]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Predict

### Bug

hello every one
i am trying to create labels for my large image datasets using the pretrained yolov8n model
i have 2 issues 
first is that when i run this code:
`model = YOLO('yolov8n.pt')

results = model("/home/Tohidi/Yolo/Datasets/train/images", stream=False, device="cuda", save_txt=True, classes=0,batch=1000, verbose=False)  `

the kernel crashes every time, its not an issue with the batch size as i am running it on A100
i dont know why is that and i want to fix that

the other issue is that in the same code when i turn the stream argument into True, the code doesnt save the txt label files although the save_txt is steel set to True.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
Hello! 😊 Let's address your issues one by one:

1. For the kernel crashing issue: It's curious that this is happening even on an A100 GPU. It might not be directly related to the batch size, but potentially to how the data is being loaded or how memory is managed. Could you please check if your system has enough memory available to handle the process? Sometimes, pre-fetching a large number of images can cause memory spikes. Additionally, reviewing any error logs can provide more clues.

2. Regarding the `stream=True` not saving .txt label files: When `stream=True` is enabled, YOLO operates in a mode optimized for video and continuous stream processing, where saving to .txt might not be directly supported due to how data flows in this mode. However, one workaround could be iterating through the results manually and saving the required data to .txt files. Here's a small example:

```python
from ultralytics import YOLO
import numpy as np

# Load model and run prediction with stream=True
model = YOLO('yolov8n.pt')
results = model("/home/Tohidi/Yolo/Datasets/train/images", stream=True, device="cuda", classes=0)

# Manually save to .txt
for i, result in enumerate(results):
    np.savetxt(f'labels/{i}.txt', result.boxes.numpy(), fmt="%d")
```

Note: Ensure the `labels/` directory exists or adjust the saving path as needed. This code snippet iteratively saves the detection boxes to .txt for each frame or image processed in the stream.

I hope this helps! If these suggestions don't resolve your issues, it might be beneficial to provide more details on the error messages you're encountering, especially for the kernel crash problem.

**Comment by mohamad-tohidi** [comment]
wow thank you very much
my server has 125 GB of ram memory
the reason i wanted to set the stream to True was to keep the kernel from crashing because of the large amount of images
but thank you very much
now i can run it easily

**Comment by glenn-jocher** [comment]
You're welcome! 😊 I'm glad to hear the response was helpful in getting you started, even if it wasn't perfectly accurate. If you have any more questions or need further assistance down the line, feel free to reach out. Happy coding and good luck with your project!

**Comment by mgupta70** [comment]
@mohamad-tohidi To enable save_txt without running into memory issues, process images in small batches.

`stream=True` does not work with `model.predict`.

So, if you have say an `image_dir` which you want to process, I would do something like this:

```python
image_paths = glob(os.path.join(image_dir, "*.jpg"))  # Adjust for your file format

# Process images in batches
batch_size = 16  # Adjust batch size as per your system's capacity
output_dir = "path/to/output/segmentation_results"  # Directory to save results
os.makedirs(output_dir, exist_ok=True)  # Ensure output directory exists

for i in range(0, len(image_paths), batch_size):
    batch_paths = image_paths[i:i + batch_size]  # Get a batch of image paths

    # Run prediction on the batch
    results = model.predict(
        source=batch_paths,
        save_txt=True,        # Save predictions as .txt files
        save=False,           # Disable saving annotated images (optional)
        project=output_dir,   # Output directory
        name="predictions",          
        exist_ok=True         # Overwrite to use the same directory and avoid creating new folders for each batch
    )

```

Hope this helps!
